src/Filter.php

```diff
@@ -872,6 +879,13 @@ public function apply($query)
             throw new UnauthorizedException();
         }
 
+        if (!empty($this->per)) {
+            $query->getModel()->setPerPage($this->value);
```

Comment:
This is not how other filters work. 

Use withQuery and remove the class property $per